libcontainer/cgroups/fs/cpuset.go

```diff
@@ -48,7 +50,109 @@ func (s *CpusetGroup) Remove(d *cgroupData) error {
 	return removePath(d.path("cpuset"))
 }
 
+func getNumericStat(path string, filename string, extracted *[]uint16) error {
+	fileContent, err := fscommon.GetCgroupParamString(path, filename)
+	if err != nil {
+		return err
+	}
+	if len(fileContent) == 0 {
+		return fmt.Errorf("%s found to be empty", filepath.Join(path, filename))
+	}
+
+	for _, s := range strings.Split(fileContent, ",") {
+		if strings.Contains(s, "-") {
+			splitted := strings.Split(s, "-")
```

Comment:
nit: `strings.Split(s, "-", 3)` so you won't split more than necessary. We use `3` to detect extra `-`.

Reply:
If it can perform more than one split in this place, it means that file is corrupted and should error out.

Reply:
right, that is what I meant.
1. Use `strings.SplitN(s, "-", 3)` instead of `strings.Split(s, "-")` to avoid potential extra splitting that you don't need.
2. Since you use `3` you can still detect the corrupted input by checking for `len(splitted) != 2`

Reply:
@piowag ^^^

Reply:
This has been fixed